src/spatialdata/_io/_utils.py

```diff
@@ -282,6 +288,57 @@ def save_transformations(sdata: SpatialData) -> None:
     """
     from spatialdata.transformations import get_transformation, set_transformation
 
-    for element in sdata._gen_elements_values():
+    for element in sdata._gen_spatial_element_values():
         transformations = get_transformation(element, get_all=True)
         set_transformation(element, transformations, set_all=True, write_to_sdata=sdata)
+
+
+def read_table_and_validate(
+    zarr_store_path: str, group: zarr.Group, subgroup: zarr.Group, tables: dict[str, AnnData]
+) -> dict[str, AnnData]:
+    """
+    Read in tables in the tables Zarr.group of a SpatialData Zarr store.
+
+    Parameters
+    ----------
+    zarr_store_path
+        The path to the Zarr store.
+    group
+        The parent group containing the subgroup.
+    subgroup
+        The subgroup containing the tables.
+    tables
+        A dictionary of tables.
+
+    Returns
+    -------
+    The modified dictionary with the tables.
+    """
+    count = 0
+    for table_name in subgroup:
+        f_elem = subgroup[table_name]
+        f_elem_store = os.path.join(zarr_store_path, f_elem.path)
+        if isinstance(group.store, zarr.storage.ConsolidatedMetadataStore):
+            tables[table_name] = read_elem(f_elem)
+            # we can replace read_elem with read_anndata_zarr after this PR gets into a release (>= 0.6.5)
+            # https://github.com/scverse/anndata/pull/1057#pullrequestreview-1530623183
+            # table = read_anndata_zarr(f_elem)
```

Comment:
is this comment still valid @LucaMarconato , I am not sure.

Reply:
I think the latest AnnData handles this, gonna open a issue on this and we can do it in a small PR.

Reply:
Opened: https://github.com/scverse/spatialdata/issues/418. Tiny code fix but need to be tested on remote objects, I'd do it separately.